app/service-accounts/server/methods/usernameExists.js

```diff
@@ -0,0 +1,18 @@
+import { Meteor } from 'meteor/meteor';
+import { check } from 'meteor/check';
+
+import { Users } from '../../../models';
+
+Meteor.methods({
+	usernameExists(name) {
```

Comment:
You might want to add rate-limiter to this method

Reply:
When a user is creating a new service account, this is to show which username is not available before submission. Should I put a 10 calls per user per hour limit?

Reply:
There is an already existing implementation of this, which also includes check for blocklisted usernames. Can you update the code to use that?
See app/lib/server/functions/checkUsernameAvailability.js

Reply:
I have updated this method with this function.